include/deal.II/differentiation/ad/ad_drivers.h

```diff
@@ -622,39 +786,176 @@ namespace Differentiation
     {
       using scalar_type = double;
 
+      TapedDrivers()
+        : active_tape(numbers::invalid_tape_index)
+        , keep_values(true)
```

Comment:
This and many functions below could well be moved into a .cc file, and maybe should -- this would make the header file substantially simpler.

I'm ok if you put this onto your todo list and do this in a follow-up patch. It doesn't have to happen as part of the current PR.

Reply:
See #7335